cpp/src/umap/runner.h

```diff
@@ -91,7 +91,8 @@ void _fit(const cumlHandle &handle,
 
   if (params->verbose)
     std::cout << "n_neighbors=" << params->n_neighbors << std::endl;
-  find_ab(params, d_alloc, stream);
+
+  if (params->a == -1. || params->b == -1.) find_ab(params, d_alloc, stream);
```

Comment:
Very small thing- but this logic actually shouldn't be in the C++ `_fit()` anymore, as it's already being called from the constructor at the Python layer. This `if` statement should be done in Python and the call to `find_ab()` removed from the c ++ `fit()`